Modules/IO/JPEG/src/itkJPEGImageIO.cxx

```diff
@@ -506,29 +504,26 @@ JPEGImageIO::WriteSlice(std::string & fileName, const void * const buffer)
   // start compression
   jpeg_start_compress(&cinfo, TRUE);
 
-  volatile const JSAMPLE * outPtr = ((const JSAMPLE *)buffer);
-
-  // write the data. in jpeg, the first row is the top row of the image
-  auto *    row_pointers = new JSAMPROW[height];
-  const int rowInc = numComp * width;
-  for (unsigned int ui = 0; ui < height; ++ui)
   {
-    row_pointers[ui] = const_cast<JSAMPROW>(outPtr);
-    outPtr = const_cast<JSAMPLE *>(outPtr) + rowInc;
+    const auto * ptr = static_cast<const JSAMPLE *>(buffer);
+    const auto   rowbytes = num_comp * width;
+    for (size_t ui = 0; ui < height; ++ui)
+    {
+      row_pointers[ui] = const_cast<JSAMPLE *>(ptr) + rowbytes * ui;
+    }
   }
-  jpeg_write_scanlines(&cinfo, row_pointers, height);
-
-  if (fflush(fp) == EOF)
```

Comment:
I have check, fflush is called [here](https://github.com/InsightSoftwareConsortium/ITK/blob/a89145bccda6a36f42cfdd45d3a6b27234ff54fe/Modules/ThirdParty/JPEG/src/itkjpeg/jdatadst.c#L173) in `term_destination (j_compress_ptr cinfo) `called by `jpeg_finish_compress(&cinfo);`